File: AIDataQuery.API/Models/Entities/QueryLog.cs
using AIDataQuery.API.Models.Enums;

namespace AIDataQuery.API.Models.Entities;

public class QueryLog
{
    public long Id { get; set; }
    public int UserId { get; set; }
    public string? PlatformCode { get; set; }
    public string? DatabaseName { get; set; }
    public string SqlContent { get; set; } = string.Empty;
    public int ExecutionTimeMs { get; set; }
    public int RowCount { get; set; }
    public QueryStatus Status { get; set; }
    public string? ErrorMessage { get; set; }
    public string? ClientIp { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public User? User { get; set; }
}

File: AIDataQuery.Web/src/types/queryLog.ts
import { QueryStatus, QueryParams } from './api'

// 查询日志相关类型

/** 查询日志DTO */
export interface QueryLogDto {
  id: number
  username: string
  platformCode?: string
  databaseName?: string
  sqlContent: string
  executionTimeMs: number
  rowCount: number
  status: QueryStatus
  errorMessage?: string
  clientIp?: string
  createdAt: string
}

/** 查询日志筛选参数 */
export interface QueryLogParams extends QueryParams {
  platformCode?: string
  status?: QueryStatus
  startDate?: string
  endDate?: string
}
